subprojects/model-core/src/main/java/org/gradle/model/internal/core/NodeBackedModelSet.java

```diff
@@ -71,17 +71,22 @@ public String toString() {
     }
 
     @Override
-    public void create(final Action<? super T> action) {
+    public void create(Action<? super T> action) {
+        create(getElementTypeClass(), action);
+    }
+
+    @Override
+    public <S extends T> void create(Class<S> type, final Action<? super T> action) {
         state.assertCanMutate();
 
-        String name = String.valueOf(modelNode.getLinkCount(ModelNodes.withType(elementType)));
+        String name = String.valueOf(modelNode.getLinkCount(ModelNodes.withType(type)));
```

Comment:
I guess node paths aren't entirely clear to me yet. Which classes should I look at to get a better grasp of paths? I'd be happy to spruce up the documentation if I can grok the subject.